### ISSUE TITLE
FR: Optionally purge Keyvault Managed HSMs

### ISSUE THREAD

**Issue by jackofallops** [tombuildsstuff/azurerm-dalek#6]
Managed HSMs do not belong to a Resource Group, and are billed (along with the KV) for in soft deleted state after the Key Vault to which they belong is removed for the remaining retention period, which is 90 days by default and has a minimum of 7 days.

These can be found and purged manually (provided Purge Protection is not enabled) via:
```
az keyvault list-deleted --resource-type hsm
az keyvault purge --hsm-name [hsmname] --location [location]
```
(appropriate login and selection of `--subscription` req'd)

We should add an option to clear any of these found as this can be a significant cost, and by default there can only be 5 Managed HSM pools in a subscription per region.

> Error: Code="ManagedHsmInsufficentQuota" Message="You may only create only 5 HSM pool(s) per subscription in a region."

**Comment by tombuildsstuff** [comment]
Further to this I think we're going to need to also support some additional methods here, including also breaking the pairing on any servicebus namespaces etc - we'll likely want to implement this in a manner supporting multiple, given there's already a few use cases for this popping up

**Comment by tombuildsstuff** [comment]
Types:

* Removing Locks
* Purging HSMs within a given Location
* Breaking the pairing for a ServiceBus namespace
* Backup related

**Comment by tombuildsstuff** [comment]
Fixed via #9
